reactor-core/src/jcstress/java/reactor/core/publisher/FluxReplayStressTest.java

```diff
@@ -0,0 +1,684 @@
+					"concurrency when synchronize block is entered, the connection is " +
+					"nulled so this will cause a connect() to be called. Current " +
+					"ConnectableFlux api does not allow any improvements on that since " +
+					"it lacks coordination.")
+	@Outcome(id = {"[0,1], 1, [0,1], 0, 0, 1, [1,2]"}, expect = ACCEPTABLE_INTERESTING,desc =
+			"Expected corner case when second subscriber still join the first " +
+					"subscription but due to natural concurrency cancellation happens " +
+					"before onComplete is called so the second subscriber gets value " +
+					"and error instead of onComplete. second connect call may still " +
+					"happen since ConnectableFlux.subscribe happens before the check of" +
+					" current connection value in FluxRefCnt, thus different ")
+	@Outcome(id = {"0, 0, 0, 0, 0, 1, 2"}, expect = ACCEPTABLE_INTERESTING,desc =
+			"Expected corner case when second subscriber still join the first " +
+					"subscription but due to natural concurrency cancellation of the " +
+					"first subscriber happens before value is delivered so in that case" +
+					" error is delivered prior any values")
+	@State
+	public static class RefCntRaceSubscribeAndCancelNoTimeoutStressTest {
+
+		final LongAdder     sourceSubscriptionsCnt = new LongAdder();
+		final Flux<Integer> sharedSource;
+
+		final StressSubscriber<Integer> subscriber1 = new StressSubscriber<>();
+		final StressSubscriber<Integer> subscriber2 = new StressSubscriber<>();
+
+		public RefCntRaceSubscribeAndCancelNoTimeoutStressTest() {
+			this.sharedSource = Flux.range(0, 1)
+			                        .doOnSubscribe(s -> sourceSubscriptionsCnt.increment())
+			                        .replay(1)
+			                        .refCount(1, Duration.ofMillis(0));
+		}
+
+		@Actor
+		public void subscribe1() {
+			sharedSource.subscribe(subscriber1);
+		}
+
+		@Actor
+		public void subscribe2() {
+			sharedSource.subscribe(subscriber2);
+		}
+
+		@Actor
+		public void cancel() {
+			subscriber1.cancel();
+		}
+
+		@Arbiter
+		public void arbiter(IIIIIII_Result r) {
+			r.r1 = subscriber1.onNextCalls.get();
+			r.r2 = subscriber2.onNextCalls.get();
+			r.r3 = subscriber1.onCompleteCalls.get();
+			r.r4 = subscriber2.onCompleteCalls.get();
+			r.r5 = subscriber1.onErrorCalls.get();
+			r.r6 = subscriber2.onErrorCalls.get();
+			r.r7 = sourceSubscriptionsCnt.intValue();
+		}
+	}
+
+	@JCStressTest
+	@Outcome(id = {
+			"100000, 100000"}, expect = ACCEPTABLE, desc = "concurrent subscription succeeded")
+	@State
+	public static class RefCntRaceManySubscribersSubscribeAndCancelNoTimeoutStressTest
+			extends RefCntConcurrentSubscriptionBaseStressTest<Integer> {
+
+		final LongAdder signalCount1 = new LongAdder();
+		final LongAdder signalCount2 = new LongAdder();
+
+		public RefCntRaceManySubscribersSubscribeAndCancelNoTimeoutStressTest() {
+			super(Flux.range(0, 1), 1, Duration.ofMillis(0));
+		}
+
+		/*
```

Comment:
@OlegDokuka are you planning to arrange these comments still? If not I can do so.

Reply:
Aligned the comments.